### DISCUSSION TITLE
Stream url mp3 to speaker 100w ( Amp built_in)

### DISCUSSION THREAD

**Discussion by vplong990** [pschatzmann/arduino-audio-tools#286] [Q&A] [answered]
I can stream mp3 (128kbps) to ESP32 with PCM5102A and 5w speaker no problem, but when I tried to use 100W speaker (amp built in) , the music seem delay short time ( ~ 0.1s) in during stream. Sorry my bad English.

Respectfully

**Comment by pschatzmann** [comment, marked as answer]
See
https://github.com/pschatzmann/arduino-audio-tools/wiki/Performance
I would expect that mp3 is adding more delay then wav due to the internal buffering.

Note that I don't know how much delay your hardware components are adding.
